upgrade_agent/upgrade_agent_handlers.c

```diff
@@ -3,28 +3,42 @@
 #include <img_utils.h>
 #include <flash_utils.h>
 #include <upgrade_agent.h>
+#include <version.h>
 
 static struct app_info_block *last_known_info;
+extern uint32_t _flash_start;
+extern uint32_t _flash_end;
+
+static int offset_in_bootloader(uint32_t offset)
+{
+	return !!((offset < (uint32_t)&_flash_end) &&
```

Comment:
If using `!!` here then why not use a boolean type instead?  Unless you are not using C99 or later.